projects/bzip2/bzip2_compress_target.c

```diff
@@ -0,0 +1,67 @@
+/*
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+###############################################################################
+*/
+
+#include "bzlib.h"
+#include <stdint.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+
+extern int BZ2_bzBuffToBuffCompress(char* dest,
+                           unsigned int* destLen,
+                           char*         source,
+                           unsigned int  sourceLen,
+                           int           blockSize100k,
+                           int           verbosity,
+                           int           workFactor);
+
+extern int BZ2_bzBuffToBuffDecompress(char* dest,
+                                      unsigned int* destLen,
+                                      char*         source,
+                                      unsigned int  sourceLen,
+                                      int           small,
+                                      int           verbosity);
+
+int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    int r;
+    unsigned int nZ, nOut;
+    char *zbuf = malloc(size + 600 + (size / 100));
```

Comment:
What are these hardcoded values? Are they guaranteed to always work?

Reply:
They have been borrowed from [here](https://github.com/google/bzip2-rpc/blob/master/unzcrash.c#L42). I am not sure if it is always guaranteed to work, but this code is present in a test harness that is required to test the robustness of (de)compression. 

@julian-seward1 what do you think?

Reply:
I think it's OK.  They just reflect the fact that the worst case output size is 101% of the input size + 600 bytes (I assume -- this is now nearly 20 years old).  Since the buffer is in mallocville, presumably asan will complain if it gets overrun.  I doubt that will happen though.